### PR TITLE
github: update PR template to require testing + make changelog notice seen by reviewers

### LINKED ISSUE #2312: Update PR template to require testing
1. Require PR authors to add tests, or provide a strong argument as to why they didn't and how they plan to test otherwise.
2. Require reviewers to sign off on the PR having appropriate tests or sufficient reason to not add them (and that they vouch for the stability of the change).

### PR DESCRIPTION
**Pull request by slimsag**
Fixes #2312

I have intentionally made this short and concise. The point is to make both the author and reviewer think about updating the changelog and verifying the change is well tested, but not to be over-prescriptive (i.e., I want both the author and reviewer to think about these -- but not prescribe the exact course of action that needs to be taken).

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1 +1,2 @@
-<!-- Remember to update the changelog for user-facing changes. -->
+- [ ] I've updated the changelog (or do not need to).
+- [ ] This change is covered by tests, or I have a strong reason for not doing so (explain).
```
Comment (keegancsmith) [inline .github/pull_request_template.md]: I like how Phabricator does this, can I suggest we follow suite:

```
<!-- PR Description -->

<!-- Reminder: Have you updated the changelog? -->

Test Plan: <!-- insert required test plan here -->
```

I also think things like the changelog being part of the PR description is noisy, and should only be a reminder to the author. Which is why I put it as a comment in my example above.

Comment (slimsag) [reply .github/pull_request_template.md]: The problem with comments is that only the author can see them.

I also want to remind reviewers on PRs that they should check for proper testing and that the PR updated the changelog appropriately.

Unfortunately, GitHub doesn't support anything like [a reviewer template](https://github.community/t5/How-to-use-Git-and-GitHub/Pull-Request-Review-Template/td-p/6857), so the best I can do is make these checklist items that are visible to the reviewer.

Any better suggestions on how I can achieve that? (reminding both the author *and* reviewer that testing + changelog updates should be done)

Comment (keegancsmith) [reply .github/pull_request_template.md]: Does the reviewer need to care about changelog entries? I feel like the reminder as a comment is quite effective, or have you noticed authors ignoring it?

The test plan is visible to both and the reviewer should review the test plan.

Comment (slimsag) [reply .github/pull_request_template.md]: Fair argument. 👍 I'll go your route for now (but may remove `<!-- PR Description -->` so people don't have to delete that line), and we can see how that works.

**Pull request merged**